src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,273 @@
+        return new Iterator<T>() {
+
+            private Node<T> lastReturned;
+            private Node<T> next = first;
+            private int nextIndex = 0;
+
+            @Override
+            public boolean hasNext() {
+                return size > nextIndex;
+            }
+
+            @Override
+            public T next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                } else {
+                    lastReturned = next;
+                    next = next.next;
+                    nextIndex++;
+                }
+                return lastReturned.item;
+            }
+
+            @Override
+            public void remove() {
+                if (lastReturned == null)
+                    throw new IllegalStateException();
+
+                Node<T> lastNext = lastReturned.next;
+                if (next == lastReturned)
+                    next = lastNext;
+                else
+                    nextIndex--;
+                lastReturned = null;
+            }
+        };
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        Object[] result = new Object[size];
+        int i = 0;
+
+        Iterator<T> iter = iterator();
+        while (iter.hasNext()) {
+            result[i] = iter.next();
+            i++;
+        }
+
+        return result;
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
-    }
+        if (a.length < size)
+            a = (T1[]) java.lang.reflect.Array.newInstance(
+                    a.getClass().getComponentType(), size);
+        int i = 0;
+        Object[] result = a;
+        for (Node<T> x = first; x != null; x = x.next)
+            result[i++] = x.item;
 
-    public boolean add(T t) {
-        return false;
+        if (a.length > size)
+            a[size] = null;
+
+        return a;
     }
 
     public boolean remove(Object o) {
+        if (o == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (o.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
+    T unlink(Node<T> x) {
+        final T element = x.item;
+        final Node<T> next = x.next;
+        final Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.item = null;
+        size--;
+        return element;
+    }
+
     public boolean containsAll(Collection<?> c) {
```

Comment:
Should check if your range contains all elements from collection c. Review you logic.

Reply:
If not ALL elements are present then return false.